### PR TITLE
Feature/create mail service

### PR DESCRIPTION
**Pull request by Anatolii-Omelchenko**

### CODE REVIEW

File: src/main/java/baza/trainee/domain/dto/MailDto.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import static baza.trainee.constants.MailConstants.FIELD_MESSAGE_ERROR_MSG;
+import static baza.trainee.constants.MailConstants.FIELD_NAME_ERROR_MSG;
+import static baza.trainee.constants.MailConstants.MAX_MSG_LENGTH;
+import static baza.trainee.constants.MailConstants.MAX_NAME_LENGTH;
+import static baza.trainee.constants.MailConstants.MIN_MSG_LENGTH;
+import static baza.trainee.constants.MailConstants.MIN_NAME_LENGTH;
+import static baza.trainee.constants.MailConstants.NULL_FIELD_ERROR_MSG;
+
+public record MailDto(
+        @NotBlank(message = NULL_FIELD_ERROR_MSG)
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/domain/dto/MailDto.java:18]: Blank line before annotation

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        emailTemplate = emailTemplate.replace("{{firstName}}", firstName);
+        emailTemplate = emailTemplate.replace("{{lastName}}", lastName);
+        emailTemplate = emailTemplate.replace("{{email}}", email);
+        emailTemplate = emailTemplate.replace("{{message}}", message);
+
+        return emailTemplate;
+    }
+
+    private String readHtmlTemplateFromFile(final String filePath) throws IOException {
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: Would it be better to catch the IOException in the service layer and throw a specified runtime exception?

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        emailTemplate = emailTemplate.replace("{{firstName}}", firstName);
+        emailTemplate = emailTemplate.replace("{{lastName}}", lastName);
+
+        return emailTemplate;
+    }
+
+    @Override
+    public String buildMsgForMuseum(final String firstName, final String lastName,
+                                    final String email, final String message) throws IOException {
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: Would it be better to catch the IOException in the service layer and throw a specified runtime exception?

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            helper.setFrom(MUSEUM_EMAIL, MUSEUM_LABEL);
+            helper.setSubject(subject);
+            mailSender.send(mimeMessage);
+        } catch (MessagingException | UnsupportedEncodingException ex) {
+            throw new EmailSendingException(FAIL_SEND_MSG);
+        }
+    }
+
+    @Override
+    public String buildMsgForUser(final String firstName, final String lastName) throws IOException {
```
Comment (malyshevhen) [inline src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: Would it be better to catch the IOException in the service layer and throw a specified runtime exception?

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        emailTemplate = emailTemplate.replace("{{firstName}}", firstName);
+        emailTemplate = emailTemplate.replace("{{lastName}}", lastName);
+        emailTemplate = emailTemplate.replace("{{email}}", email);
+        emailTemplate = emailTemplate.replace("{{message}}", message);
+
+        return emailTemplate;
+    }
+
+    private String readHtmlTemplateFromFile(final String filePath) throws IOException {
+        ClassPathResource resource = new ClassPathResource(filePath);
```
Comment (Olga232) [inline src/main/java/baza/trainee/service/impl/MailServiceImpl.java:78]: What about using Thymeleaf to process templates?

File: src/main/java/baza/trainee/constants/MailConstants.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package baza.trainee.constants;
+
+public final class MailConstants {
+    private MailConstants() {
+    }
+
+    public static final String MUSEUM_EMAIL = "museum@gmail.com";
```
Comment (Olga232) [inline src/main/java/baza/trainee/constants/MailConstants.java]: Should we set these parameters (MUSEUM_EMAIL, MUSEUM_LABEL, MUSEUM_TEMPLATE_PATH, USER_TEMPLATE_PATH) in application.yml?

File: src/main/java/baza/trainee/domain/dto/MailDto.java (hunk 1 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/domain/dto/MailDto.java:18]: fixed

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java (hunk 2 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: fixed

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java (hunk 3 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: fixed

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java (hunk 4 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/service/impl/MailServiceImpl.java]: fixed

File: src/main/java/baza/trainee/constants/MailConstants.java (hunk 6 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/constants/MailConstants.java]: fixed

File: src/main/java/baza/trainee/service/impl/MailServiceImpl.java (hunk 5 above)
Comment (Anatolii-Omelchenko) [reply src/main/java/baza/trainee/service/impl/MailServiceImpl.java:78]: resolved

**Review by Olga232 — APPROVED** [review]

**Review by malyshevhen — APPROVED** [review]

**Review by OleksandrKorkach — APPROVED** [review]

**Pull request merged**